yt/utilities/lib/cyoctree.pyx

```diff
@@ -1,679 +1,756 @@
-                                    self.nodes[i].right_edge[0],
-                                    self.nodes[i].right_edge[1],
-                                    self.nodes[i].right_edge[2],
-                                    self.nodes[i].start,
-                                    self.nodes[i].end,
-                                    self.nodes[i].parent,
-                                    self.nodes[i].children,
-                                    self.nodes[i].leaf,
-                                    self.nodes[i].node_id,
-                                    self.nodes[i].leaf_id,
-                                    self.nodes[i].depth))
-
-    def load(self, fname = None):
-        if fname is None:
-            raise ValueError("A filename must be specified to load the octtree!")
-        # clear any current tree we have loaded
-        self.reset()
-
-        cdef Node temp
-        cdef np.int64_t num_octs
-        with open(fname,'rb') as f:
-            (self._num_particles, num_octs, self._n_ref,
-             self.over_refine_factor, self.density_factor,
-             self._data_version) = \
-                struct.unpack('2Q3iq', f.read(40))
-            self._idx = \
-                np.asarray(struct.unpack('{}Q'.format(self.num_particles),
-                           f.read(8*self.num_particles)), dtype=np.int64)
-
-            reserve(&self.nodes, num_octs+1)
-            for i in range(num_octs):
-                (temp.left_edge[0], temp.left_edge[1], temp.left_edge[2],
-                 temp.right_edge[0], temp.right_edge[1], temp.right_edge[2],
-                 temp.start, temp.end, temp.parent, temp.children, temp.leaf,
-                 temp.node_id, temp.leaf_id, temp.depth) = \
-                struct.unpack('6d4Q?2QB', f.read(105))
-                self.nodes.push_back(temp)
-
-        for i in range(3):
-            self._left_edge[i] = self.nodes[0].left_edge[i]
-            self._right_edge[i] = self.nodes[0].right_edge[i]
+                                    hsml[i], prefactor, prefactor_norm,
+                                    0, use_normalization=use_normalization)
+
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-cdef int split_helper(int start, int end, int num_children, int max_splits,
-                       np.int64_t * split_arr, np.float64_t * pos,
-                       np.int64_t * idx, np.float64_t * left,
-                       np.float64_t * right) nogil except -1:
-    '''
-    This takes in the split array and sets up the first and last particle, it
-    then calls the spit function.
-    '''
-    cdef double lower = left[0]
-    cdef double upper = right[0]
-
-    split_arr[0] = start
-    split_arr[num_children] = end
-
-    split(0, num_children, max_splits, 0, 0, split_arr, idx, pos, left, right,
-          lower, upper)
+cdef int seperate_x(Octree * octree, double value, np.int64_t start, np.int64_t end) nogil except -1:
```

Comment:
Shouldn't this be `separate_x`?
In addition the function is very similar to separate_{yz}. Maybe you could refactor as
```cython
cdef inline int _separate_helper(np.float64_t * pos, np.int64_t * pidx, double value, const np.int64_t start, np.int64_t end) nogil except -1:
    cdef int index
    for index in range(start, end):
        if pos[pidx[index]] < value:
            pidx[split], pidx[index] = pidx[index], pidx[split]
            split += 1

    return split


cdef int separate_x(Octree * octree, double value, np.int64_t start, np.int64_t end) nogil except -1:
    """docstring"""
    return _separate_helper(octree.pposx, octree.pidx, value, start, end)


cdef int separate_y(Octree * octree, double value, np.int64_t start, np.int64_t end) nogil except -1:
    """docstring"""
    return _separate_helper(octree.pposy, octree.pidx, value, start, end)


cdef int separate_z(Octree * octree, double value, np.int64_t start, np.int64_t end) nogil except -1:
    """docstring"""
    return _separate_helper(octree.pposz, octree.pidx, value, start, end)
```